lib/request.js

```diff
@@ -14,37 +12,48 @@ const version = require("../package.json").version;
  * @returns {Promise}
  */
 function request(baseConfig, method, depth, xml) {
-    const config = _.assign({}, baseConfig);
-
-    config.headers = _.assign({}, baseConfig.headers);
+    const config = {
+        headers: baseConfig.headers || {},
+        body: xml,
+        method,
+        'headers["Content-length"]': xml.length,
+        'headers["Depth"]': depth,
```

Comment:
This will not add the headers correctly to the config. We should add them directly to the headers object, like this: